src/components/timeline/TimelineMarkerCanvas.js

```diff
@@ -297,14 +330,36 @@ class TimelineMarkerCanvas extends PureComponent {
     y: CssPixels,
     width: CssPixels,
     height: CssPixels,
-    cornerSize: CssPixels
+    radius: CssPixels
   ) {
-    // Cut out c x c -sized squares in the corners.
-    const c = Math.min(width / 2, Math.min(height / 2, cornerSize));
-    const bottom = y + height;
-    ctx.fillRect(x + c, y, width - 2 * c, c);
-    ctx.fillRect(x, y + c, width, height - 2 * c);
-    ctx.fillRect(x + c, bottom - c, width - 2 * c, c);
+    // stolen from https://www.rgraph.net/canvas/reference/arcto.html
+
+    const r = Math.min(radius, height / 2, width / 2);
+    const x2 = width - 1;
+    const y2 = height - 1;
+
+    // Save the existing state of the canvas so that it can be restored later
+    ctx.save();
+
+    // Translate to the given X/Y coordinates
+    ctx.translate(x, y);
+
+    ctx.beginPath();
+
+    // Move to the center of the top horizontal line
+    ctx.moveTo(x2 / 2, 0);
+
+    // Draw the rounded corners. The connecting lines in between them are drawn automatically
+    ctx.arcTo(x2, 0, x2, y2, Math.min(y2 / 2, r));
+    ctx.arcTo(x2, y2, 0, y2, Math.min(x2 / 2, r));
+    ctx.arcTo(0, y2, 0, 0, Math.min(y2 / 2, r));
+    ctx.arcTo(0, 0, radius, 0, Math.min(x2 / 2, r));
+
+    // Draw a line back to the start coordinates
+    ctx.lineTo(x2 / 2, 0);
+    ctx.fill();
+
+    ctx.restore();
```

Comment:
with the previous implementation, when zooming in (I mean the browser zoom) they were gaps inside the rectangle because of the way it was done (with 3 fillRect). Now this is a lot cleaner.

TBH I didn't find this myself -- I initially tried but failed, and I still don't understand properly how this works ;)